GitUI/BranchTreePanel/RepoObjectsTree.cs

```diff
@@ -163,34 +164,49 @@ public async Task ReloadAsync()
                 treeMain.BeginUpdate();
                 _rootNodes.ForEach(t => t.IgnoreSelectionChangedEvent = true);
                 var tasks = _rootNodes.Select(r => r.ReloadAsync(token)).ToArray();
-                await Task.WhenAll(tasks).ConfigureAwait(false);
+                await Task.WhenAll(tasks).ConfigureAwait(true);
             }
             finally
             {
-                await this.SwitchToMainThreadAsync();
+                ThreadHelper.AssertOnUIThread();
+
                 if (!token.IsCancellationRequested)
                 {
                     Enabled = true;
+                }
+
+                // Need to end the update for the selected node to scroll into view below under certain conditions
+                // (e.g. when the treeview gets larger, trying to make the selected node visible doesn't work).
+                treeMain.EndUpdate();
+
+                if (!token.IsCancellationRequested)
+                {
                     var selectedNode = treeMain.AllNodes().FirstOrDefault(n =>
                         _rootNodes.Any(rn => $"{rn.TreeViewNode.FullPath}{treeMain.PathSeparator}{currentBranch}" == n.FullPath));
                     if (selectedNode != null)
                     {
-                        treeMain.SelectedNode = selectedNode;
-                        ExpandPathToSelectedNode();
+                        SetSelectedNode(selectedNode);
                     }
 
                     _rootNodes.ForEach(t => t.IgnoreSelectionChangedEvent = false);
                 }
-
-                treeMain.EndUpdate();
             }
+        }
 
-            void ExpandPathToSelectedNode()
-            {
-                treeMain.Scrollable = false; // disable scrolling, so the next call does not horizontally (nor vertically) scroll
-                treeMain.SelectedNode.EnsureVisible();
-                treeMain.Scrollable = true;
-            }
+        private const int WM_HSCROLL = 276;
+        private const int SB_LEFT = 6;
+        [DllImport("user32.dll")]
+        private static extern int SendMessage(IntPtr hwnd, int wmsg, int wparam, int lparam);
```

Comment:
We have `GitUI.NativeMethods` class that has a definition for `SendMessage` method.
Could we use that one?

In fact we have p/invokes sprinkled all over the place, and we should consolidate them in one place.

Reply:
Yes, you're right. I'll make use of the NativeMethods version of SendMessageInt, and add my constants to it. We should consolidate the other calls as well, but as a separate change, perhaps?

Reply:
Yes, as a separate PR.
I merely made an observation of a tech debt we'll have to address.